app/controllers/comments_controller.rb

```diff
@@ -1,28 +1,45 @@
 class CommentsController < ApplicationController
+  before_action :article
   def create
-    @article = Article.find(params[:article_id])
-    @comment = @article.comments.create(comment_params)
-    redirect_to article_path(@article)
+    @comment = article.comments.new(comment_params) do |comment|
+      comment.user = current_user
+    end
+    if @comment.save
+      redirect_to article_path(article)
+    else 
+      redirect_to article_path(article)
+    end
   end
 
   def edit
-    @article = Article.find(params[:article_id])
-    @comment = @article.comments.edit(comment_params)
+    @comment = Comment.find(params[:id])
   end
 
   def update
     @comment = Comment.find(params[:id])
 
     if @comment.update(comment_params)
-      redirect_to article_path(@article)
+      redirect_to article_path(article)
     else
-      redirect_to article_path
+      render :edit
     end
   end
 
+  def destroy
+    @comment = Comment.find(params[:id])
+    @comment.destroy
+    redirect_to article_path(article)
+   
+
+  end
+
   private
 
   def comment_params
     params.require(:comment).permit(:commenter, :body)
```

Comment:
Optional you can use `tap` here, so you can do something like this

```
params.require(:comment).permit(:commenter, :body).tap do |permitted_params|
  permitted_params[:user_id] = current_user.id
  permitted_params[:article_id] = @article.id
end
```

https://apidock.com/rails/Object/tap

Reply:
or we can also use `merge` to merge current_user to comment_params as well
```ruby
def comment_params
  params.require(:comment).permit(:commenter, :body).merge(user: current_user)
end
```